internal/controllers/machine/machine_controller.go

```diff
@@ -697,6 +700,10 @@ func (r *Reconciler) drainNode(ctx context.Context, cluster *clusterv1.Cluster,
 		log.V(3).Info("Node is unreachable, draining will use 1s GracePeriodSeconds and will ignore all Pods that have a deletionTimestamp > 1s old")
 	}
 
+	if machine.Status.Deletion.NodeDrainStartTime == nil {
```

Comment:
I would prefer if we would do this in l.388

Is there a reason we can't?

(just so that setting the condition & timestamp is next to each other, like for wait for node volume detach)

Reply:
Also it gets patched together with the condition then